racetime/views/race.py

```diff
@@ -431,6 +431,11 @@ def get_form_kwargs(self):
         kwargs = super().get_form_kwargs()
         kwargs['category'] = self.get_category()
         kwargs['can_moderate'] = kwargs['category'].can_moderate(self.user)
+        for field in models.Race._meta.get_fields():
+            if field.name in self.request.GET:
+                kwargs['initial'][field.name] = self.request.GET[field.name]
+                if field.name == 'custom_goal':
+                    kwargs['initial']['goal'] = ''
```

Comment:
This logic should be moved to the `CreateRace` class as well because it would potentially cause confusion if allowed on the race edit form, there's no need for that. I don't think it's sensible to apply it to `OAuthCreateRace` either.